ouroboros-network/protocol-tests/Test/ChainProducerState.hs

```diff
@@ -217,28 +216,28 @@ instance Arbitrary ChainProducerStateForkTest where
   arbitrary = do
     TestChainFork _ c f <- arbitrary
     let l = Chain.length c
-    rs <- Map.fromList . zip [0..] <$> listOf (genReaderState l c)
+    rs <- Map.fromList . zip [0..] <$> listOf (genFollowerState l c)
     return $ ChainProducerStateForkTest (ChainProducerState c rs (length rs)) f
 
   shrink (ChainProducerStateForkTest (ChainProducerState c rs nr) f)
-    -- shrink readers
+    -- shrink followers
      = [ ChainProducerStateForkTest (ChainProducerState c rs' nr) f
        | rs' <- map Map.fromList . shrinkList (const []) . Map.toList $ rs
        ]
     -- shrink the fork chain
     ++ [ ChainProducerStateForkTest (ChainProducerState c rs nr) f'
        | TestBlockChain f' <- shrink (TestBlockChain f)
        ]
-    -- shrink chain and fix up readers
-    ++ [ ChainProducerStateForkTest (ChainProducerState c' (fixupReaderPointer c' <$> rs) nr) f
+    -- shrink chain and fix up followers
+    ++ [ ChainProducerStateForkTest (ChainProducerState c' (fixupFollowerPointer c' <$> rs) nr) f
        | TestBlockChain c' <- shrink (TestBlockChain c)
        ]
     where
-      fixupReaderPointer :: Chain Block -> ReaderState Block -> ReaderState Block
-      fixupReaderPointer c' r@ReaderState{readerPoint} =
-        if pointOnChain readerPoint c'
+      fixupFollowerPointer :: Chain Block -> FollowerState Block -> FollowerState Block
+      fixupFollowerPointer c' r@FollowerState{followerPoint} =
```

Comment:
`r@FollowerState`. We agree that finding all of the `r`-to-`f` translations was likely untenable, but the `r@F` are pretty easy to catch. Let's get those. (This isn't the only one, but they're easy to grep for: `git grep -e 'r@F' -- '*hs'`.)

Reply:
Thanks! The recent commit fixes it. Please also see the commit message on why I've changed `r` to `fs`